Tasks/AzureFunctionOnKubernetesV0/src/deploy.ts

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+import * as yaml from 'js-yaml';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as tr from "azure-pipelines-task-lib/toolrunner";
+import { CommandHelper } from './utils/commandHelper';
+import { DockerConnection } from './dockerConnection';
+import { Kubectl, Resource } from 'kubernetes-common-v2/kubectl-object-model';
+import * as FileHelper from './utils/fileHelper';
+import * as KubernetesConstants from 'kubernetes-common-v2/kubernetesconstants';
+import * as KubernetesManifestUtility from 'kubernetes-common-v2/kubernetesmanifestutility';
+import * as CommonUtils from 'kubernetes-common-v2/utility';
+
+const secretName = tl.getInput('secretName');
+const appName = tl.getInput('appName', true);
+const namespace = getKubernetesNamespace();
+const dockerHubNamespace = tl.getInput('dockerHubNamespace');
+const args = tl.getInput('arguments');
+const funcRootDir = tl.getInput('functionRootDirectory', true);
+const waitForStability = tl.getBoolInput('waitForStability');
+const isDryRun = (args && args.includes('--dry-run')) ? true : false;
```

Comment:
Group them into a data class maybe?

Reply:
Currently inputs are read in deploy only. Will change to that model if the task expands.